Code/Tools/AssetProcessor/native/utilities/AssetServerHandler.cpp

```diff
@@ -60,21 +116,120 @@ namespace AssetProcessor
+            if (jsonBuffer.empty())
+            {
+                // no configuration to save
+                return;
+            }
+
+            // save the configuration
+            rapidjson::Document recognizerDoc;
+            recognizerDoc.Parse(jsonBuffer.c_str());
+            AZ::JsonSerializationUtils::WriteJsonFile(recognizerDoc, settingsFilePath.LexicallyNormal().c_str());
+        }
+        else if (m_assetCachingMode == AssetServerMode::Client)
+        {
+            // load the configuration
+            if (!AZ::IO::SystemFile::Exists(settingsFilePath.c_str()))
+            {
+                // no log since it is okay to not have a settings file
+                return;
+            }
+
+            auto result = AZ::JsonSerializationUtils::ReadJsonFile(settingsFilePath.LexicallyNormal().c_str());
+            if (!result.IsSuccess())
+            {
+                AZ_Warning(AssetProcessor::DebugChannel, false, "ACS settings file failed with (%s)", result.GetError().c_str());
+                return;
+            }
+
+            rapidjson::StringBuffer stringBuffer;
+            rapidjson::Writer<rapidjson::StringBuffer> writer(stringBuffer);
+            if (result.GetValue().Accept(writer) == false)
+            {
+                AZ_Warning(AssetProcessor::DebugChannel, false, "ACS failed to load settings file (%s)", settingsFilePath.c_str());
+                return;
+            }
+
+            RecognizerContainer recognizerContainer;
+            if (!AssetProcessor::PlatformConfiguration::ConvertFromJson(stringBuffer.GetString(), recognizerContainer))
+            {
+                AZ_Warning(AssetProcessor::DebugChannel, false, "ACS failed to convert settings file (%s)", settingsFilePath.c_str());
+                return;
+            }
+
+            recognizerConfiguration->AddAssetCacheRecognizerContainer(recognizerContainer);
+        }
+    }
+
+    AssetServerMode AssetServerHandler::GetRemoteCachingMode() const
+    {
+        return m_assetCachingMode;
+    }
+
+    void AssetServerHandler::SetRemoteCachingMode(AssetServerMode mode)
+    {
+        m_assetCachingMode = mode;
+        AssetServerNotificationBus::Broadcast(&AssetServerNotificationBus::Events::OnRemoteCachingModeChanged, mode);
+    }
+
+    const AZStd::string& AssetServerHandler::GetServerAddress() const
+    {
+        return m_serverAddress;
+    }
+
+    void AssetServerHandler::SetServerAddress(const AZStd::string& address)
+    {
+        AZStd::string previousServerAddress = m_serverAddress;
+        m_serverAddress = address;
+        if (!IsServerAddressValid())
+        {
+            m_serverAddress = previousServerAddress;
+            AZ_Error(AssetProcessor::DebugChannel, false,
+                "Server address (%s) is invalid! Reverting back to (%s)",
+                address.c_str(),
+                previousServerAddress.c_str());
```

Comment:
you could use %.*s and AZ_STRING_ARG(previousServerAddress) for AZStd::string printouts. I don't know if you want to change it, it doesn't really make the code more or less readable.

Reply:
all of the cool kids are doing it :)